pkg/image/registry/imagestream/strategy.go

```diff
@@ -156,12 +156,6 @@ func (s Strategy) tagsChanged(old, stream *api.ImageStream) fielderrors.Validati
 		}
 
 		if tagRef.From.Kind == "DockerImage" && len(tagRef.From.Name) > 0 {
-			event, err := tagReferenceToTagEvent(stream, tagRef, "")
-			if err != nil {
-				errs = append(errs, fielderrors.NewFieldInvalid(fmt.Sprintf("spec.tags[%s].from", tag), tagRef.From, err.Error()))
-				continue
-			}
-			api.AddTagEventToImageStream(stream, tag, *event)
```

Comment:
Ok, just wanted to be clear.  I think the challenge is that there will be
repos we can't import (because we don't have auth yet, or it's down) - if
that blocks an admin from directly tagging that's bad.  On the other hand,
trying to get the image metadata is really really important.  So there
still has to be a bypass import flow.

On Wed, Sep 30, 2015 at 3:59 PM, Andy Goldstein <notifications@github.com>
wrote:

> In pkg/image/registry/imagestream/strategy.go
> <https://github.com/openshift/origin/pull/4853#discussion_r40796195>:
>
> > @@ -156,12 +156,6 @@ func (s Strategy) tagsChanged(old, stream *api.ImageStream) fielderrors.Validati
> >  		}
> >
> >  		if tagRef.From.Kind == "DockerImage" && len(tagRef.From.Name) > 0 {
> > -			event, err := tagReferenceToTagEvent(stream, tagRef, "")
> > -			if err != nil {
> > -				errs = append(errs, fielderrors.NewFieldInvalid(fmt.Sprintf("spec.tags[%s].from", tag), tagRef.From, err.Error()))
> > -				continue
> > -			}
> > -			api.AddTagEventToImageStream(stream, tag, *event)
>
> I think the description was confusing. With this PR, when you set
> spec.tags[2.2].from=openshift/ruby-22-centos7 (kind=DockerImage), you only
> get *1* image:
>
>   - items:
>     - created: 2015-09-30T13:49:37Z
>       dockerImageReference: openshift/ruby-20-centos7@sha256:509c5a47b689e1523388ff6c160b0ea1417f412794b4a9854277b272de7384d3
>       image: sha256:509c5a47b689e1523388ff6c160b0ea1417f412794b4a9854277b272de7384d3
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4853/files#r40796195>.
>